### ISSUE TITLE
[MU4 Issue] Crash in Score::endCmd due to unsafe use of global variable

### ISSUE THREAD

**Issue by wizofaus** [musescore/MuseScore#8364]
**Describe the bug**
Slightly random, haven't been able to reproduce 100% consistently, but certain actions (in this case trying to change the duration of a note) can cause a crash in Score::endCmd, specifically at the line `MuseScoreCore::mscoreCore->endCmd(isCmdFromInspector, rollback)`.
This seems like quite dangerous and unnecessary use of a global variable. At the point it crashed `MuseScoreCore::mscoreCore` appeared to refer to something already deleted.

**To Reproduce**
1. Create a new score
2. Open the new score dialog (Ctrl+N)
3. Cancel/close dialog
4. Add a note
5. MuseScore sometimes crashes (sometimes)

**Expected behavior**
Preferably, applicaton should not crash!

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

read access violation
>	MuseScore4.exe!Ms::Score::endCmd(const bool isCmdFromInspector, bool rollback) Line 303	C++
 	MuseScore4.exe!mu::notation::NotationUndoStack::commitChanges() Line 110	C++
 	MuseScore4.exe!mu::notation::NotationNoteInput::padNote(const Ms::Pad & pad) Line 202	C++
 	MuseScore4.exe!mu::notation::NotationActionController::padNote(const Ms::Pad & pad) Line 497	C++

Virtual table pointer for MuseScoreCore::mscoreCore was invalid.

Basically ~MuseScoreCore can occasionally be called for the object currently referred to by MuseScoreCore::mscoreCore but I haven't managed to work out why/when it happens.

**Comment by wizofaus** [comment]
Happy to fix - except that I don't really understand why `MuseScoreCore::mscoreCore` is a global variable at all, or what it's for?

**Comment by igorkorsukov** [comment]
> Happy to fix - except that I don't really understand why `MuseScoreCore::mscoreCore` is a global variable at all, or what it's for?

For historical reasons.
libmscore was taken from MU3 almost as it is.
MU3 uses a lot of global variables, I don’t know why, probably because they didn’t know how to do it differently, there was no Dependency Injection system, and so on.
Step by step we will get rid of all global variables

**Comment by wizofaus** [comment]
PluginAPI seems to be the only thing that actually relies on it being a global variable as far as I can tell, and I don't believe the endCmd() and updateInspector() functions are needed at all.  I'm happy to rip it all out as long as there's no intention to use them for something...
